source/adios2/toolkit/remote/Remote.cpp

```diff
@@ -54,8 +54,8 @@ void Remote::InitCMData()
 {
     std::lock_guard<std::mutex> lockGuard(m_CMInitMutex);
     bool first = true;
-    auto &CM = CManagerSingleton::Instance(first);
-    ev_state.cm = CM.m_cm;
+    auto CM = CManagerSingleton::Instance(first);
+    ev_state.cm = CM->m_cm;
```

Comment:
So, this is singleton magic.  It is a raw pointer, but every instance is to the same raw pointer and it is deallocated on exit by normal C++ mechanisms.  So, no leak, and yes it is run through CI and this test even goes through msan successfully (we don't run all tests in msan because it's too slow).  However, there's something slightly off about the singleton somehow, because it's supposed to be thread safe, yet when the file-level remote stuff creates multiple transports it doesn't get past tsan.  So this PR is limited to getting the get-level remote stuff tested and I'll sort out the problems with the thread safety in the next PR.